src/main/java/com/mappy/fpm/batches/tomtom/helpers/RoadTagger.java

```diff
@@ -161,6 +161,38 @@ public static void addTagIf(String key, String value, boolean condition, Map<Str
         }
     }
 
+    public void tagConstruction(Map<String, String> tags){
+        if (tags.containsKey("construction")) {
+
+            switch (tags.get("construction")) {
+                case "both":
+                    tags.put("construction", tags.get("highway"));
+                    break;
+                case "forward":
+                    if (tags.containsKey("oneway") && tags.get("oneway").equals("yes")){
+                        tags.put("construction", tags.get("highway"));
+                    }
+                    else{
+                        tags.put("construction:forward", tags.get("highway"));
+                        tags.remove("construction");
+                    }
+                    break;
+                case "backward":
+                    if (tags.containsKey("oneway") && tags.get("oneway").equals("yes")){
+                        tags.put("construction", tags.get("highway"));
+                    }
+                    else{
+                        tags.put("construction:backward", tags.get("highway"));
+                        tags.remove("construction");
+                    }
+                    break;
+                default:
+                    break;
+            }
```

Comment:
It would maybe be a bit cleaner if you separate the logic that choses the proper tag, and the effective replacement/set of the `construction` tag.